src/Dapr.Cryptography/Encryption/DaprEncryptionGrpcClient.cs

```diff
@@ -0,0 +1,188 @@
+    [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public override async Task<ReadOnlyMemory<byte>> DecryptAsync(
+        string vaultResourceName,
+        ReadOnlyMemory<byte> ciphertextBytes,
+        string keyName,
+        DecryptionOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        using var memoryStream = ciphertextBytes.CreateMemoryStream(true);
+        var decryptionResult = DecryptAsync(vaultResourceName, memoryStream, keyName, options, cancellationToken);
+
+        var bufferedResult = new ArrayBufferWriter<byte>();
+        await foreach (var item in decryptionResult)
+        {
+            bufferedResult.Write(item.Span);
+        }
+
+        return bufferedResult.WrittenMemory;
+    }
+
+    /// <summary>
+    /// Decrypts the specified stream of ciphertext using the Dapr Cryptography encryption functionality.
+    /// </summary>
+    /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+    /// <param name="ciphertextStream">The stream containing the bytes of the ciphertext value to decrypt.</param>
+    /// <param name="keyName">The name of the key to use from the Vault for the decryption operation.</param>
+    /// <param name="options">Options informing how the decryption operation should be configured.</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+    /// <returns>An asynchronously enumerable array of decrypted bytes.</returns>
+    [Obsolete(
+        "The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public override async IAsyncEnumerable<ReadOnlyMemory<byte>> DecryptAsync(
+        string vaultResourceName,
+        Stream ciphertextStream,
+        string keyName,
+        DecryptionOptions? options = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+        ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+        ArgumentVerifier.ThrowIfNull(ciphertextStream, nameof(ciphertextStream));
+        options = options ?? new DecryptionOptions();
+        
+        var decryptRequestOptions = new Autogenerated.DecryptRequestOptions
+        {
+            ComponentName = vaultResourceName, 
+            KeyName = keyName
+        };
+
+        var grpcCallOptions = DaprClientUtilities.ConfigureGrpcCallOptions(typeof(DaprEncryptionClient).Assembly,
+            this.DaprApiToken, cancellationToken);
+        var duplexStream = Client.DecryptAlpha1(grpcCallOptions);
+
+        using var streamProcessor = new DecryptionStreamProcessor();
+        await streamProcessor.ProcessStreamAsync(ciphertextStream, duplexStream, options.StreamingBlockSizeInBytes,
```

Comment:
This await doesn't really do anything given how the processor is setup.

I think a design like you see in the Azure.Data.Tables library would be amenable here: https://learn.microsoft.com/en-us/dotnet/api/azure.data.tables.tableclient.queryasync?view=azure-dotnet

Basically, instead of returning a `Task` the process stream returns a wrapper which includes the IAsyncEnumerable.

Reply:
It's true that await doesn't really do anything, but between it and just discarding the value, C# requires that I put something there. I don't know that I want to return any additional public types just for the sake of it when I could just return the `IAsyncEnumerable` and leave it to consumers to decide what to do with it.